src/Core/Settings.h

```diff
@@ -592,6 +592,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(UInt64, function_range_max_elements_in_block, 500000000, "Maximum number of values generated by function 'range' per block of data (sum of array sizes for every row in a block, see also 'max_block_size' and 'min_insert_block_size_rows'). It is a safety threshold.", 0) \
     M(ShortCircuitFunctionEvaluation, short_circuit_function_evaluation, ShortCircuitFunctionEvaluation::ENABLE, "Setting for short-circuit function evaluation configuration. Possible values: 'enable' - use short-circuit function evaluation for functions that are suitable for it, 'disable' - disable short-circuit function evaluation, 'force_enable' - use short-circuit function evaluation for all functions.", 0) \
     \
+    M(String, storage_file_read_method, "mmap", "Method of reading data from storage file, one of: read, pread, mmap, pread_threadpool.", 0) \
```

Comment:
It would be better to use enum here instead of String, so this value will be checked when we set it, because otherwise we can set some random value and get an exception only when using engine `file`. See file `SettingsEnum.h`. We can reuse `LocalFSReadMethod` there and declare only 3 possible values for this setting `mmap/pread/read`.